--- pyo30.29.0/Cargo.toml
+++ pyo30.29.2/Cargo.toml
@@ -13,7 +13,7 @@
 edition = "2021"
 rust-version = "1.83"
 name = "pyo3"
-version = "0.29.0"
+version = "0.29.2"
 authors = ["PyO3 Project and Contributors <https://github.com/PyO3>"]
 build = "build.rs"
 links = "pyo3-python"
@@ -488,10 +488,10 @@
 optional = true
 
 [dependencies.pyo3-ffi]
-version = "=0.29.0"
+version = "=0.29.2"
 
 [dependencies.pyo3-macros]
-version = "=0.29.0"
+version = "=0.29.2"
 optional = true
 
 [dependencies.rust_decimal]
@@ -567,7 +567,7 @@
 features = ["v4"]
 
 [build-dependencies.pyo3-build-config]
-version = "=0.29.0"
+version = "=0.29.2"
 
 [target.'cfg(not(target_arch = "wasm32"))'.dev-dependencies.ctrlc]
 version = "3.5.2"
@@ -608,6 +608,8 @@
 check-cfg = [
     "cfg(pyo3_disable_reference_pool)",
     "cfg(pyo3_leak_on_drop_without_reference_pool)",
+    "cfg(pyo3_use_raw_dylib)",
+    'cfg(pyo3_dll, values("python3", "python3_d", "python3t", "python3t_d", "python38", "python38_d", "python39", "python39_d", "python310", "python310_d", "python311", "python311_d", "python312", "python312_d", "python313", "python313_d", "python313t", "python313t_d", "python314", "python314_d", "python314t", "python314t_d", "python315", "python315_d", "python315t", "python315t_d", "python316", "python316_d", "python316t", "python316t_d", "libpypy3.11-c"))',
 ]
 
 [lints.rustdoc]
