--- pyo3-ffi0.29.0/src/cpython/objimpl.rs
+++ pyo3-ffi0.29.2/src/cpython/objimpl.rs
@@ -29,6 +29,16 @@
 
     #[cfg(Py_3_9)]
     pub fn PyObject_IS_GC(o: *mut PyObject) -> c_int;
+
+    #[cfg(all(Py_3_9, not(any(PyPy, GraalPy))))]
+    pub fn PyObject_GET_WEAKREFS_LISTPTR(o: *mut PyObject) -> *mut *mut PyObject;
+}
+
+#[inline]
+#[cfg(not(Py_3_9))]
+pub unsafe fn PyObject_GET_WEAKREFS_LISTPTR(o: *mut PyObject) -> *mut *mut PyObject {
+    let weaklistoffset = (*Py_TYPE(o)).tp_weaklistoffset;
+    o.byte_offset(weaklistoffset) as *mut *mut PyObject
 }
 
 #[inline]
@@ -49,10 +59,4 @@
     ((*t).tp_weaklistoffset > 0) as c_int
 }
 
-#[inline]
-pub unsafe fn PyObject_GET_WEAKREFS_LISTPTR(o: *mut PyObject) -> *mut *mut PyObject {
-    let weaklistoffset = (*Py_TYPE(o)).tp_weaklistoffset;
-    o.offset(weaklistoffset) as *mut *mut PyObject
-}
-
 // skipped PyUnstable_Object_GC_NewWithExtraData
