--- pyo30.29.0/src/internal/pyclass_init.rs
+++ pyo30.29.2/src/internal/pyclass_init.rs
@@ -177,7 +177,7 @@
     }
 }
 
-/// Analagous to `Into<PyClassInitializer<T>>`, but just an internal equivalent
+/// Analogous to `Into<PyClassInitializer<T>>`, but just an internal equivalent
 /// to avoid allowing user code to define custom return types from `#[new]`.
 trait IntoPyClassInitializer<T: PyClass>: Sized {
     fn into_pyclass_initializer(self) -> PyClassInitializer<T>;
