--- pyo30.29.0/src/conversions/std/osstr.rs
+++ pyo30.29.2/src/conversions/std/osstr.rs
@@ -1,6 +1,3 @@
-// TODO https://github.com/PyO3/pyo3/issues/5487
-#![allow(clippy::undocumented_unsafe_blocks)]
-
 use crate::conversion::IntoPyObject;
 #[cfg(not(target_os = "wasi"))]
 use crate::ffi;
@@ -50,6 +47,7 @@
             let bytes = self.as_bytes();
             let ptr = bytes.as_ptr().cast();
             let len = bytes.len() as ffi::Py_ssize_t;
+            // SAFETY: passing valid pointer to python API
             unsafe {
                 // DecodeFSDefault automatically chooses an appropriate decoding mechanism to
                 // parse os strings losslessly (i.e. surrogateescape most of the time)
@@ -62,6 +60,7 @@
         #[cfg(windows)]
         {
             let wstr: Vec<u16> = self.encode_wide().collect();
+            // SAFETY: passing valid pointer to python API
             unsafe {
                 // This will not panic because the data from encode_wide is well-formed Windows
                 // string data
@@ -130,6 +129,7 @@
 
             // Get an owned allocated wide char buffer from PyString, which we have to deallocate
             // ourselves
+            // SAFETY: passing valid pointer to python API
             let size =
                 unsafe { ffi::PyUnicode_AsWideChar(pystring.as_ptr(), core::ptr::null_mut(), 0) };
             crate::err::error_on_minusone(ob.py(), size)?;
@@ -141,6 +141,7 @@
             let size = size - 1; // exclude null terminator
 
             let mut buffer = vec![0; size as usize];
+            // SAFETY: passing valid pointer to python API
             let bytes_read =
                 unsafe { ffi::PyUnicode_AsWideChar(pystring.as_ptr(), buffer.as_mut_ptr(), size) };
             assert_eq!(bytes_read, size);
