--- pyo3-build-config0.29.0/Cargo.toml.orig
+++ pyo3-build-config0.29.2/Cargo.toml.orig
@@ -1,6 +1,6 @@
 [package]
 name = "pyo3-build-config"
-version = "0.29.0"
+version = "0.29.2"
 description = "Build configuration for the PyO3 ecosystem"
 authors = ["PyO3 Project and Contributors <https://github.com/PyO3>"]
 keywords = ["pyo3", "python", "cpython", "ffi"]
@@ -22,9 +22,5 @@
 
 # deprecated
 resolve-config = []
-
-# deprecated
 extension-module = []
-
-# deprecated: no longer needed, raw-dylib is used instead
 generate-import-lib = []
