--- pyo3-macros-backend0.29.0/src/module.rs
+++ pyo3-macros-backend0.29.2/src/module.rs
@@ -528,7 +528,7 @@
 
     let mut result = quote! {
         #[doc(hidden)]
-        pub const __PYO3_NAME: &'static ::std::ffi::CStr = #pyo3_name;
+        pub static __PYO3_NAME: &'static ::std::ffi::CStr = #pyo3_name;
 
         // This structure exists for `fn` modules declared within `fn` bodies, where due to the hidden
         // module (used for importing) the `fn` to initialize the module cannot be seen from the #module_def
@@ -544,19 +544,20 @@
                 #pyo3_path::impl_::trampoline::module_exec(module, #module_exec)
             }
 
-            // The full slots, used for the PyModExport initializaiton
-            static SLOTS: impl_::PyModuleSlots = impl_::PyModuleSlotsBuilder::new()
+            static DOC: &'static ::std::ffi::CStr = #doc;
+            static SLOTS: impl_::PrimaryModuleSlots = impl_::PyModuleSlotsBuilder::new()
                 .with_mod_exec(__pyo3_module_exec)
                 .with_abi_info()
                 .with_gil_used(#gil_used)
                 .with_name(__PYO3_NAME)
-                .with_doc(#doc)
+                .with_doc(DOC)
                 .build();
+            static SECONDARY_SLOTS: impl_::SecondaryModuleSlots = impl_::secondary_slots(&SLOTS);
 
             // Since the macros need to be written agnostic to the Python version
             // we need to explicitly pass the name and docstring for PyModuleDef
-            // initializaiton.
-            impl_::ModuleDef::new(__PYO3_NAME, #doc, &SLOTS)
+            // initialization.
+            impl_::ModuleDef::new(__PYO3_NAME, DOC, &SLOTS, &SECONDARY_SLOTS)
         };
     };
     if !is_submodule {
