--- pyo3-macros-backend0.29.0/src/pyimpl.rs
+++ pyo3-macros-backend0.29.2/src/pyimpl.rs
@@ -407,6 +407,10 @@
                 // We cant to keep the first argument type, hence this hack
                 spec.signature.arguments.pop();
                 spec.signature.python_signature.positional_parameters.pop();
+                // the `CompareOp` parameter is gone; keep the positional-only count in range
+                spec.signature
+                    .python_signature
+                    .make_all_parameters_positional_only();
                 method_introspection_code(
                     &spec,
                     attrs,
@@ -488,9 +492,13 @@
     }
     let return_type = if spec.python_name == "__new__" {
         // Hack to return Self while implementing IntoPyObject
-        parse_quote!(-> #pyo3_path::PyClassGuard<Self>)
+        // TODO: use typing.Self?
+        PyExpr::from_return_type(parse_quote!(#pyo3_path::PyClassGuard<Self>), Some(parent))
     } else {
-        spec.output.clone()
+        match spec.output.clone() {
+            ReturnType::Type(_, t) => PyExpr::from_return_type(*t, Some(parent)),
+            ReturnType::Default => PyExpr::none(),
+        }
     };
     function_introspection_code(
         pyo3_path,
