--- Git-fa647f742d7baefe8eb1181899d9c81b41559772
+++ sdist-1.99.0
@@ -12,7 +12,7 @@
 [dependencies]
 litellm-core = { workspace = true, features = ["bedrock-auth"] }
 litellm-ai-gateway = { workspace = true, default-features = false }
-pyo3 = { workspace = true, features = ["extension-module"] }
+pyo3 = { workspace = true, features = ["extension-module", "abi3-py310"] }
 pyo3-async-runtimes.workspace = true
 serde_json.workspace = true
 tokio.workspace = true
