--- pyo30.29.0/src/ffi_ptr_ext.rs
+++ pyo30.29.2/src/ffi_ptr_ext.rs
@@ -1,6 +1,3 @@
-// TODO https://github.com/PyO3/pyo3/issues/5487
-#![allow(clippy::undocumented_unsafe_blocks)]
-
 use crate::sealed::Sealed;
 use crate::{
     ffi,
@@ -41,48 +38,80 @@
 }
 
 impl FfiPtrExt for *mut ffi::PyObject {
+    /// # Safety
+    ///
+    /// see requirements for [`Bound::from_owned_ptr_or_err`]
     #[inline]
     unsafe fn assume_owned_or_err(self, py: Python<'_>) -> PyResult<Bound<'_, PyAny>> {
+        // SAFETY: caller upholds requirements
         unsafe { Bound::from_owned_ptr_or_err(py, self) }
     }
 
+    /// # Safety
+    ///
+    /// see requirements for [`Bound::from_owned_ptr_or_opt`]
     #[inline]
     unsafe fn assume_owned_or_opt(self, py: Python<'_>) -> Option<Bound<'_, PyAny>> {
+        // SAFETY: caller upholds requirements
         unsafe { Bound::from_owned_ptr_or_opt(py, self) }
     }
 
+    /// # Safety
+    ///
+    /// see requirements for [`Bound::from_owned_ptr`]
     #[inline]
     #[track_caller]
     unsafe fn assume_owned(self, py: Python<'_>) -> Bound<'_, PyAny> {
+        // SAFETY: caller upholds requirements
         unsafe { Bound::from_owned_ptr(py, self) }
     }
 
+    /// # Safety
+    ///
+    /// see requirements for [`Bound::from_owned_ptr_unchecked`]
     #[inline]
     unsafe fn assume_owned_unchecked(self, py: Python<'_>) -> Bound<'_, PyAny> {
+        // SAFETY: caller upholds requirements
         unsafe { Bound::from_owned_ptr_unchecked(py, self) }
     }
 
+    /// # Safety
+    ///
+    /// see requirements for [`Borrowed::from_ptr_or_err`]
     #[inline]
     unsafe fn assume_borrowed_or_err<'a>(
         self,
         py: Python<'_>,
     ) -> PyResult<Borrowed<'a, '_, PyAny>> {
+        // SAFETY: caller upholds requirements
         unsafe { Borrowed::from_ptr_or_err(py, self) }
     }
 
+    /// # Safety
+    ///
+    /// see requirements for [`Borrowed::from_ptr_or_opt`]
     #[inline]
     unsafe fn assume_borrowed_or_opt<'a>(self, py: Python<'_>) -> Option<Borrowed<'a, '_, PyAny>> {
+        // SAFETY: caller upholds requirements
         unsafe { Borrowed::from_ptr_or_opt(py, self) }
     }
 
+    /// # Safety
+    ///
+    /// see requirements for [`Borrowed::from_ptr`]
     #[inline]
     #[track_caller]
     unsafe fn assume_borrowed<'a>(self, py: Python<'_>) -> Borrowed<'a, '_, PyAny> {
+        // SAFETY: caller upholds requirements
         unsafe { Borrowed::from_ptr(py, self) }
     }
 
+    /// # Safety
+    ///
+    /// see requirements for [`Borrowed::from_ptr_unchecked`]
     #[inline]
     unsafe fn assume_borrowed_unchecked<'a>(self, py: Python<'_>) -> Borrowed<'a, '_, PyAny> {
+        // SAFETY: caller upholds requirements
         unsafe { Borrowed::from_ptr_unchecked(py, self) }
     }
 }
